test/Microsoft.AspNetCore.Mvc.FunctionalTests/DnxRoslynCompilationService.cs

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if DNX451 || DNXCORE50
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.PortableExecutable;
+#if DNXCORE50
+using System.Runtime.Loader;
+#endif
+using System.Text;
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Dnx.Compilation.CSharp;
+using Microsoft.Extensions.CompilationAbstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    /// <summary>
+    /// A type that uses Roslyn to compile C# content and <see cref="ILibraryExporter"/> to find out references.
+    /// </summary>
+    public class DnxRoslynCompilationService : ICompilationService
```

Comment:
This is used for running tests from VS \ `dnx test` in command line for dnxcore50 tfm. I'll follow up with another PR that switches to the CLI test runner which removes dnx dependencies in command line builds.